### ISSUE TITLE
An error occurred during installation

### ISSUE THREAD

**Issue by upchair** [elastic/windows-installers#189]
In trying to install elastic search version 6.2.4 it failed to started the elasticsearch service and gave the following error:

"An error occurred during installation."

I must say that initially when I installed elasticsearch about 3 weeks ago, i installed it successfully as a service and it worked. However I noticed that intermittently it stopped working. However when I restart my machine the service used to start up successfully.

However about 3 days ago the same problem occurred again (service stopped working). Since then I have uninstalled it and re-installed it. However the problem is still the same. The elasticsearch service has refused to start at all. 

Please find attached below the installation log file. 

[MSIe4dcc.LOG](https://github.com/elastic/windows-installers/files/1964839/MSIe4dcc.LOG)

Can anyone please help me resolve this issue. 

Thank You 

Richard

**Comment by russcam** [comment]
@upchair Looking in the log, the Elasticsearch service failed to start successfully:
```
MSI (s) (34:58) [19:22:47:535]: Product: Elasticsearch 6.2.4 -- Error 1920. Service 'Elasticsearch' (Elasticsearch) failed to start.  Verify that you have sufficient privileges to start system services.
```
I can also see that the data, logs and config directories from the previous installation at 

```
- DataDirectory = C:\ProgramData\Elastic\Elasticsearch\data
- ConfigDirectory = C:\ProgramData\Elastic\Elasticsearch\config
- LogsDirectory = C:\ProgramData\Elastic\Elasticsearch\logs
```

have been picked up and used:

```
MSI (s) (34!F8) [19:18:14:290]: PROPERTY CHANGE: Adding ConfigDirectoryExists property. Its value is 'True'.
MSI (s) (34!F8) [19:18:14:291]: PROPERTY CHANGE: Adding LogsDirectoryExists property. Its value is 'True'.
MSI (s) (34!F8) [19:18:14:291]: PROPERTY CHANGE: Adding DataDirectoryExists property. Its value is 'True'.
```
Would you be able to provide the contents of the elasticsearch.log file in `C:\ProgramData\Elastic\Elasticsearch\logs`?

If you don't want to keep any of the data from the previous installation, would you be able to delete DataDirectory, ConfigDirectory and LogsDirectory and try the installation again?

**Comment by upchair** [comment]
@russcam Thank you for your reply. Please find attached the elasticsearch.log file you requested.

Thank You 

Richard

[elasticsearch.log](https://github.com/elastic/windows-installers/files/1966381/elasticsearch.log)

**Comment by russcam** [comment]
Thanks @upchair. Nothing useful in the log sadly. 

> If you don't want to keep any of the data from the previous installation, would you be able to delete DataDirectory, ConfigDirectory and LogsDirectory and try the installation again?

Would you be able to try this?

**Comment by upchair** [comment]
@russcam Ok. I will do that and get back to you in a few minutes. 

Thanks

Richard

**Comment by upchair** [comment]
@russcam . Thank you for your help. I have resolved the problem. The problem was with the JAVA_HOME path in the system environment variable GUI. 

Apparently the java directory name had changed from jre1.8.0_161 to jre1.8.0_171 due to an update by java. This must have happened the day I tried to start  elasticsearch as a service but failed to start.

When the jave directory name changed  due to a java update as explained above, this name change was unfortunately not reflected in the JAVA_HOME path in the system environment variable which was still had a path name as follows:

**C:\Program Files\Java\jre1.8.0_161\**    (Old Version Name)

INSTEAD  OF

**C:\Program Files\Java\jre1.8.0_171\**    (New Version Name)


This of course resulted in a JAVA_HOME path conflict.


The solution is to let the new java_home system environment variable REFLECT the same name as the new JAVA directory name.

After I made the correction,elasticsearch started up as a service succesfully once again.

Thank you once again for your help. I really appreciate it.
